pandas/_libs/groupby.pyx

```diff
@@ -145,12 +147,14 @@ def group_median_float64(ndarray[float64_t, ndim=2] out,
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
-def group_cumprod_float64(float64_t[:, ::1] out,
-                          const float64_t[:, :] values,
-                          const intp_t[::1] labels,
-                          int ngroups,
-                          bint is_datetimelike,
-                          bint skipna=True) -> None:
+def group_cumprod_float64(
+    float64_t[:, ::1] out,
```

Comment:
@realead i stumbled on your `indirect_buffer` repo and thought you might have some advice on further optimizing `some_type[:, ::1] out` in cases where `out` was created with `np.empty(shape, dtype=whatever)`.  Can we get extra mileage with something like `view.indirect_contiguous`?

(Just commented on this PR bc the diff included a line with the syntax of interest)

Reply:
> in cases where `out` was created with `np.empty(shape, dtype=whatever)`. Can we get extra mileage with something like `view.indirect_contiguous`?

IIIC: No. Numpy arrays don't support indirect layout - their layout is always direct. indirect layout would make sense e.g. for combining multiple 1D-numpy arrays into one 2D-buffer without copying the data from the original numpy arrays. I don't see how this can be applied here - as you have pointed out, `out` will be created by `np.empty`, so it has direct layout - nothing to be improved upon.